src/main/java/backend/ParameterIFace.java

```diff
@@ -0,0 +1,83 @@
+package backend;
+
+import java.util.ArrayList;
+
+/**
+ * Provide a uniform interface for accessing sets of parameters for
+ * a given algorithm.
+ */
+public interface ParameterIFace {
+    ArrayList<WrappedParam> getParameters();
+}
+
+class ParameterIFace_NN implements ParameterIFace {
+    ArrayList<WrappedParam> params;
+    /**
+     * Creates an array of all useful parameters, their mins/maxes, flags, etc.
+     * For now, we are skipping: E,G,A,B,C,I,R,D
+     * @return
+     */
+    public ParameterIFace_NN(){
+        ArrayList<WrappedParam> params = new ArrayList<WrappedParam>();
+        //learning rate
+        WrappedParam learnRate = new WrappedParamFloat("float","learning rate","-L",0,1);
```

Comment:
It is indeed. I guess we could turn these into static finals, but without a compelling reason for it I am reluctant.